superagi/tools/github/review_pull_request.py

```diff
@@ -0,0 +1,140 @@
+import ast
+from typing import Type, Optional
+
+from pydantic import BaseModel, Field
+
+from superagi.helper.github_helper import GithubHelper
+from superagi.helper.json_cleaner import JsonCleaner
+from superagi.helper.prompt_reader import PromptReader
+from superagi.helper.token_counter import TokenCounter
+from superagi.llms.base_llm import BaseLlm
+from superagi.models.agent import Agent
+from superagi.tools.base_tool import BaseTool
+
+
+class GithubReviewPullRequestSchema(BaseModel):
+    repository_name: str = Field(
+        ...,
+        description="Repository name in which file hase to be added",
+    )
+    repository_owner: str = Field(
+        ...,
+        description="Owner of the github repository",
+    )
+    pull_request_number: int = Field(
+        ...,
+        description="Pull request number",
+    )
+
+
+class GithubReviewPullRequest(BaseTool):
+    """
+    Reviews the githun pull request and adds comments inline
+
+    Attributes:
+        name : The name.
+        description : The description.
+        args_schema : The args schema.
+    """
+    llm: Optional[BaseLlm] = None
+    name: str = "Github Review Pull Request"
+    args_schema: Type[BaseModel] = GithubReviewPullRequestSchema
+    description: str = "Add pull request for the github repository"
+    agent_id: int = None
+    agent_execution_id: int = None
+
+    def _execute(self, repository_name: str, repository_owner: str, pull_request_number: int) -> str:
+        """
+        Execute the add file tool.
+
+        Args:
+            repository_name: The name of the repository to add file to.
+            repository_owner: Owner of the GitHub repository.
+            pull_request_number: pull request number
+
+        Returns:
+            Pull request success message if pull request is created successfully else error message.
+        """
+        try:
+            github_access_token = self.get_tool_config("GITHUB_ACCESS_TOKEN")
+            github_username = self.get_tool_config("GITHUB_USERNAME")
+            github_helper = GithubHelper(github_access_token, github_username)
+
```

Comment:
we can avoid the use of github_helper as a temporary variable

Reply:
We need to pass it to next method.